pkg/collector/corechecks/cluster/ksm/kubernetes_state_aggregators.go

```diff
@@ -106,6 +138,52 @@ func (a *countObjectsAggregator) accumulate(metric ksmstore.DDMetric) {
 	a.accumulator[labelValues]++
 }
 
+func (a *lastCronJobCompleteAggregator) accumulate(metric ksmstore.DDMetric) {
+	a.aggregator.accumulate(metric, metrics.ServiceCheckOK)
+}
+
+func (a *lastCronJobFailedAggregator) accumulate(metric ksmstore.DDMetric) {
+	a.aggregator.accumulate(metric, metrics.ServiceCheckCritical)
+}
+
+func (a *lastCronJobAggregator) accumulate(metric ksmstore.DDMetric, state metrics.ServiceCheckStatus) {
+	if condition, found := metric.Labels["condition"]; !found || condition != "true" {
+		return
+	}
+	if metric.Val != 1 {
+		return
+	}
+
+	namespace, found := metric.Labels["namespace"]
+	if !found {
+		return
+	}
+
+	jobName, found := metric.Labels["job_name"]
+	if !found {
+		return
+	}
+
+	dashPosition := strings.LastIndexByte(jobName, '-')
+	if dashPosition == -1 {
+		return
+	}
+
+	cronjobName := jobName[:dashPosition]
+	cronjobID := jobName[dashPosition+1:]
```

Comment:
I was wondering if we could reuse [ParseCronJobForJob](https://github.com/DataDog/datadog-agent/blob/8419970298cae40293c6edc253a049e557decbe3/pkg/util/kubernetes/helpers.go#L46) - or maybe add a similar common helper that returns both the cronjob name and ID.

Reply:
Definitively !
Here is the change to leverage `ParseCronJobForJob`: https://github.com/DataDog/datadog-agent/pull/11139/commits/d1749e22248468ca55648a9f8188928069d99492.